compiler/rustc_codegen_llvm/src/lib.rs

```diff
@@ -178,7 +179,28 @@ impl WriteBackendMethods for LlvmCodegenBackend {
     type ThinBuffer = back::lto::ThinBuffer;
     fn print_pass_timings(&self) {
         unsafe {
-            llvm::LLVMRustPrintPassTimings();
+            let mut size = 0;
+            let cstr = llvm::LLVMRustPrintPassTimings(&mut size as *mut usize);
+            if cstr.is_null() {
+                std::io::stdout().write_all("failed to get pass timings".as_bytes()).unwrap();
```

Comment:
write_all is only necessary to skip utf8 validation below, here you can just print

```suggestion
                println!("failed to get pass timings");
```

Reply:
There's the same situation below which you did not address yet